front-end/src/pages/login/AuthProvider.jsx

```diff
@@ -0,0 +1,47 @@
+import { createContext, useContext, useEffect, useLayoutEffect, useState } from "react";
+
+const AuthContext = createContext(undefined);
+
+export const useAuth = () => {
+    const authContext = useContext(AuthContext);
+
+    if (!authContext) {
+        throw new Error('useAuth must be used within a AuthProvider');
+    }
+
+    return authContext;
+}
+
+const AuthProvider = ({children}) => {
```

Comment:
Where and what is this used for? Some comments in this file would help.

Reply:
I was testing a potential solution to the cookies issue but the code did not work in the way I was looking for. The file will be deleted in the next pr.